api/src/main/java/pl/kolendateam/dadcard/characterCard/entity/Character.java

```diff
@@ -1,29 +1,89 @@
 package pl.kolendateam.dadcard.characterCard.entity;
 
+import java.util.ArrayList;
+
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
 import lombok.Setter;
+import pl.kolendateam.dadcard.classCharacter.entity.ClassPg;
+import pl.kolendateam.dadcard.classCharacter.entity.SavingThrow;
+
 
 @NoArgsConstructor
 @Getter
 @Setter
 @Entity
-@RequiredArgsConstructor
 public class Character {
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    int id;
+    public int id;
 
     @NonNull
     String characterName;
 
     @NonNull
     String playerName;
-}
+
+    @JdbcTypeCode(SqlTypes.JSON)
+    ArrayList<ClassPg> classPgArray;
+
+    int lep;
+
+    @JdbcTypeCode(SqlTypes.JSON)
+    SavingThrow savingThrow;
+
+    public Character(String characterName, String playerName){
+        this.characterName = characterName;
+        this.playerName = playerName;
+        this.classPgArray = new ArrayList<>();
+        this.savingThrow = new SavingThrow();
+    }
+    
+    public void addClassToPgArray(ClassPg classPg) {
+        this.classPgArray.add(classPg);
+    }
+
+    public void incrementLevelClassForIndex(int index){
+        this.getClassPgArray().get(index).incrementLevel();
+    }
+
+    public void incrementLep(){
+        this.lep +=1;
+    }
+
+    public void addSTLevelOne(ClassPg classPg){
+
+        String stringSavingThrow = classPg.getSavingThrow();
+
+            double bonus;
+            if(stringSavingThrow.charAt(0) == 'h'){
+                bonus = 2.5;
+            } else{bonus = 0;}
+            this.savingThrow.setFortitude(this.savingThrow.getFortitude()+bonus);
+
```

Comment:
taka logika co do zasady nie powinna byc w encji. 
- reposyturium bierze encje i na głupa próbuje zapisać do DB
- controler przyjmuje requesty
- encja trzyma dane z DB i umie je sformatować do pobrania i zapisu.
Jeśli mamy jakieś obliczenia warunkowania to należy stworzyć serwice który wywołamy w controlerze . A w serwisie policzymy co potrzebujemy przekażemy do encji i encje do repo